### ISSUE TITLE
Can't download images (linux)

### ISSUE THREAD

**Issue by ScoreUnder** [hydrusnetwork/hydrus#51]
I'm trying to help set up the desu repository, and I've come across a rather big issue: images can't be downloaded.

https://i.imgur.com/gmBJd0x.png

After receiving all updates from a file server, from a fresh client installation, the thumbnails are downloaded but the images are not. "client_files" subdirectories contain no files, and an exception occurs whenever a thumbnail is clicked (because it's trying to create the bigger thumbnail but it doesn't have the file). There is an option in the right-click menu, "file repositories" → "download", but that doesn't appear to have any effect. I'm suspicious of the option too, because if it was allowed to not be downloaded, it wouldn't give me an exception.

Looking through the Sync code in ClientFiles I can't actually see anything that would download the images, so I'm quite confused by this.

"Copy traceback information" isn't working for me and the exception isn't in the logs either. I'll attach exception info after a little digging.

**Comment by ScoreUnder** [comment]
I got better behaviour from the exceptions by adding these lines to CatchExceptionClient:

```python
        sys.stderr.write("%r: %s\n" % (etype, value))
        traceback.print_tb(tb)
```

That printed:
```
<class 'include.HydrusExceptions.NotFoundException'>: File not found!
  File "/opt/hydrus/include/HydrusController.py", line 62, in EventPubSub
    try: HydrusGlobals.pubsub.WXProcessQueueItem()
  File "/opt/hydrus/include/HydrusPubSub.py", line 103, in WXProcessQueueItem
    if do_callable: callable( *args, **kwargs )
  File "/opt/hydrus/include/ClientGUICanvas.py", line 1282, in FocusChanged
    if page_key == self._page_key: self.SetMedia( media )
  File "/opt/hydrus/include/ClientGUICanvas.py", line 932, in SetMedia
    self._media_container = MediaContainer( self, self._image_cache, self._current_display_media, initial_size, initial_position )
  File "/opt/hydrus/include/ClientGUICanvas.py", line 2720, in __init__
    self._MakeMediaWindow()
  File "/opt/hydrus/include/ClientGUICanvas.py", line 2751, in _MakeMediaWindow
    else: self._media_window = self._media_window = StaticImage( self, self._media, self._image_cache, media_initial_size, media_initial_position )
  File "/opt/hydrus/include/ClientGUICanvas.py", line 3047, in __init__
    self.EventResize( None )
  File "/opt/hydrus/include/ClientGUICanvas.py", line 3129, in EventResize
    if self._image_container is None: self._image_container = self._image_cache.GetImage( self._media, ( my_width, my_height ) )
  File "/opt/hydrus/include/ClientCaches.py", line 334, in GetImage
    image_container = HydrusImageHandling.ImageContainer( media, target_resolution )
  File "/opt/hydrus/include/HydrusImageHandling.py", line 511, in __init__
    RasterContainer.__init__( self, media, target_resolution )
  File "/opt/hydrus/include/HydrusImageHandling.py", line 493, in __init__
    self._path = ClientFiles.GetFilePath( hash, mime )
  File "/opt/hydrus/include/ClientFiles.py", line 159, in GetFilePath
    if path is None or not os.path.exists( path ): raise HydrusExceptions.NotFoundException( 'File not found!' )
<class 'include.HydrusExceptions.NotFoundException'>: File not found!
  File "/opt/hydrus/include/ClientGUICanvas.py", line 3129, in EventResize
    if self._image_container is None: self._image_container = self._image_cache.GetImage( self._media, ( my_width, my_height ) )
  File "/opt/hydrus/include/ClientCaches.py", line 334, in GetImage
    image_container = HydrusImageHandling.ImageContainer( media, target_resolution )
  File "/opt/hydrus/include/HydrusImageHandling.py", line 511, in __init__
    RasterContainer.__init__( self, media, target_resolution )
  File "/opt/hydrus/include/HydrusImageHandling.py", line 493, in __init__
    self._path = ClientFiles.GetFilePath( hash, mime )
  File "/opt/hydrus/include/ClientFiles.py", line 159, in GetFilePath
    if path is None or not os.path.exists( path ): raise HydrusExceptions.NotFoundException( 'File not found!' )
<type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'IsRendered'
  File "/opt/hydrus/include/ClientGUICanvas.py", line 3101, in EventPaint
    self._Redraw()
  File "/opt/hydrus/include/ClientGUICanvas.py", line 3060, in _Redraw
    if self._image_container.IsRendered():
```

**Comment by hydrusnetwork** [comment]
Thank you for this report. It looks like I screwed up some location logic last week with the trash service addition, which meant the client thinks remote files are local, and so throws errors when it can't find them in the hdd. I fixed it today, so it should be better in v165.

Repo sync will only get the thumbnails--files are supposed to be downloaded manually by the user. I have plans to improve the workflow of this, because it happens behind the scenes at the moment, and user feedback on what is going on is bad. I will have a quick look at it on Tuesday and a longer look at repository use and administration all around next week.
